### PR TITLE
fix: Plumb tracing header name env/flag to client

### PR DESCRIPTION
**Pull request by mkmik**
The name of the tracing header the various IOx service us to read the tracing context value is configurable via a flag/env.

However, the flight client (e.g. used by the querier to talk to the ingester) doesn't honour that value and instead uses the hardcoded default `uber-trace-id`, which is the default jaeger header. Unfortunately I know at least one company who didn't like the name `uber-trace-id` and uses `<companyname>-trace-id` instead. For that company (which is us btw) it's impossible to see trace information produced by the ingester during the query path because the querier would send out `uber-trace-id` but the ingester will listen to `<companyname>-trace-id` and thus never read the trace context.

---

This PR plumbs the value from the config object created from the flags into the flight client, through a couple of constructor functions.

---

Tried it out both in all-in-one mode and in a testing cluster:

image

### CODE REVIEW

File: querier/src/ingester/flight_client.rs [hunk 1]
```diff
@@ -96,12 +96,20 @@ pub struct FlightClientImpl {
...
+    trace_context_header_name: String,
 }
 
 impl FlightClientImpl {
     /// Create new client.
-    pub fn new() -> Self {
-        Self::default()
+    pub fn new(trace_context_header_name: &str) -> Self {
+        //        let trace_context_header_name =
+        //            ttrace_context_header_name.to_string().to_string();
```
Comment (NGA-TRAN) [inline querier/src/ingester/flight_client.rs:108]: leftover?

**Review by NGA-TRAN — APPROVED** [review]
What is the red exclamation mark infant of the iox-ingester2? I wonder how to do unit test for this?

File: influxdb_iox/src/commands/run/querier.rs [hunk 2]
```diff
@@ -115,6 +115,11 @@ pub async fn command(config: Config) -> Result<(), Error> {
         exec,
         time_provider,
         querier_config: config.querier_config,
+        trace_context_header_name: config
+            .run_config
+            .tracing_config()
+            .traces_jaeger_trace_context_header_name
```
Comment (alamb) [inline influxdb_iox/src/commands/run/querier.rs:121]: 👍  It took me a while to figure out that the tracing config was already connected up higher up : https://github.com/influxdata/influxdb_iox/blob/main/trace_exporters/src/lib.rs#L56

and this is plumbing it down into the flight client

Nice

**Review by alamb — COMMENTED** [review]
@NGA-TRAN 's comment is a good one to resolve before merging I think

File: querier/src/ingester/flight_client.rs (hunk 1 above)
Comment (mkmik) [reply querier/src/ingester/flight_client.rs:108]: Yes

**Comment by mkmik** [comment]
> What is the red exclamation mark infant of the iox-ingester2? I wonder how to do unit test for this?

The exclamation mark means that it logs an error. I don't know the reason of that error. But if it weren't for the trace I wouldn't ever see that error since it's not logged

**Comment by mkmik** [comment]
> > What is the red exclamation mark infant of the iox-ingester2? I wonder how to do unit test for this?
> 
> The exclamation mark means that it logs an error. I don't know the reason of that error. But if it weren't for the trace I wouldn't ever see that error since it's not logged

@NGA-TRAN ah the reason is simple. When I ran this the ingester didn't have any data in memory for that table, so it returned "not-found".

I wrote some data and re-run the query and now I see something more interesting from the ingester trace:


image

**Comment by crepererum** [comment]
Wow, good detective work. IIRC this worked nicely locally (because of the default header names) and somehow this was broken in prod. Looking forward to have this link available during debugging :slightly_smiling_face:

**Pull request merged**